OpenSim/Simulation/Model/GeometryPath.cpp

```diff
@@ -136,28 +136,27 @@ generateDecorations(bool fixed, const ModelDisplayHints& hints,
         DefaultGeometry::drawPathPoint(mbix, lastPos, getColor(state), appendToThis);
 
     Vec3 pos;
-    for (int j = 1; j < pathPoints.getSize(); j++) {
-        PathPoint* point = pathPoints[j];
-        PathWrapPoint* mwp = dynamic_cast<PathWrapPoint*>(point);
-
-        if (mwp) {
-            // If the point is a PathWrapPoint and has surfacePoints,
-            // then this is the second of two tangent points for the
-            // wrap instance. So add the surface points to the display
-            // path before adding the second tangent point.
-            // Note: the first surface point is coincident with the
-            // first tangent point, so don't add it to the path.
-            Array<Vec3>& surfacePoints = mwp->getWrapPath();
-            const Transform& X_BG = mwp->getBody().getTransformInGround(state);
-            for (int j = 1; j<surfacePoints.getSize(); j++) {
-                pos = X_BG*surfacePoints.get(j);
+    int cnt = 0;
+    for (int i = 1; i < pathPoints.getSize(); ++i) {
+        PathPoint* point = pathPoints[i];
+        PathWrapPoint* pwp = dynamic_cast<PathWrapPoint*>(point);
+
+        if (pwp) {
+            // A PathWrapPoint provides points on the wrapping surface as Vec3s
+            Array<Vec3>& surfacePoints = pwp->getWrapPath();
+            // The surface points are expressed w.r.t. the wrap surface's body 
+            const Transform& X_BG = pwp->getBody().getTransformInGround(state);
+            // Cycle through each surface point and draw it
+            for (int j = 0; j<surfacePoints.getSize(); ++j) {
```

Comment:
My understanding  is that there are two "real" `PathWrapPoints` in a `GeometryPath` associated with a `WrapObject` and these are the "tangent points" where the path contacts and leaves the wrap surface. The path on the wrap surface is described by surface points (Vec3 in the WrapObject's Frame). Why this array is stuffed in one (the 2nd) of the `PathWrapPoint`'s is beyond me. The surface path apparently includes the "tangent" points and the first was not to being converted into an intermediate "PathWrapPoint" by starting the index at 1.

In the current version `PathWrapPoint` is not draw, instead only the surface points it contains are drawn. I think the new way is less confusing.